pennylane_lightning/src/util/Util.hpp

```diff
@@ -502,6 +502,132 @@ inline auto matrixVecProd(const std::vector<std::complex<T>> mat,
     return v_out;
 }
 
+/**
+ * @brief Calculates transpose of a matrix recursively and Cache-Friendly
+ * using blacking and Cache-optimized techniques.
+ */
+template <class T, size_t BLOCKSIZE = 32> // NOLINT(readability-magic-numbers)
```

Comment:
Is it possible to add `@param` docstring elements for each argument?

Reply:
Added! Please let me know if they are not coherent.